src/theory/sets/cardinality_extension.cpp

```diff
@@ -121,22 +126,42 @@ void CardinalityExtension::checkFiniteType(TypeNode& t)
   // get all equivalent classes of type t
   vector<Node> representatives = d_state.getSetsEqClasses(t);
 
-  Node typeCardinality = nm->mkConst(Rational(card.getFiniteCardinality()));
+  Node typeCardinality;
+  if (t.isInterpretedFinite())
+  {
+    typeCardinality = nm->mkConst(Rational(card.getFiniteCardinality()));
+  }
+  else
+  {
+    std::map<TypeNode, Node>::iterator skolemIt =
```

Comment:
Is this skolem necessary? Can't we just leave typeCardinality null in this case and then not send the lemma?

Reply:
Right. Nice catch 👍